src/env/tock/mod.rs

```diff
@@ -56,8 +99,75 @@ pub fn take_storage() -> StorageResult<TockStorage> {
 }
 
 impl UserPresence for TockEnv {
-    fn check(&mut self, channel: Channel) -> Result<(), Ctap2StatusCode> {
-        check_user_presence(self, channel)
+    fn check_init(&mut self, _channel: Channel) {
+        self.blink_pattern = 0;
+    }
+    fn wait_with_timeout(
+        &mut self,
+        _channel: Channel,
+        timeout: Milliseconds<ClockInt>,
+    ) -> UserPresenceResult {
+        blink_leds(self.blink_pattern);
+        self.blink_pattern += 1;
+
+        let button_touched = Cell::new(false);
+        let mut buttons_callback = buttons::with_callback(|_button_num, state| {
+            match state {
+                ButtonState::Pressed => button_touched.set(true),
+                ButtonState::Released => (),
+            };
+        });
+        let mut buttons = buttons_callback.init().flex_unwrap();
+        for mut button in &mut buttons {
+            button.enable().flex_unwrap();
+        }
+
+        let immediate_check = timeout.integer() == 0;
```

Comment:
Why not short-circuit the whole function?

```rust
fn wait_with_timeout() {
    if timeout.integer() == 0 {
        return Err(Timeout);
    }
```

Reply:
Done